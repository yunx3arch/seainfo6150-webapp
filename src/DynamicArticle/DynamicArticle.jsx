import React from "react";
import HTMLText from "../HTMLText/HTMLText";

const DynamicArticle = (props) => {
  
  return (
    <article>
      <header>
      <link to="/articlelist">Back</link>
        <h1>{props.article.title}</h1>
        <address>
          by {props.article.author} (
          <a href={props.article.authorEmail}>{props.article.authorEmail}</a>)
          <br />
        </address>
        <time dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
      </header>
      <HTMLText text={props.article.text} />
    </article>
  );
};

export default DynamicArticle;
