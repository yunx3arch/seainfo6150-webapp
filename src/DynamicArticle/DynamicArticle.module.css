.article {
    background: #fff;
    border: 1px dashed #666;
    border-radius: 10px;
    box-shadow: 0px 0px 10px 5px rgba(204, 0, 0, 0.5);
    font-family: Arial, Helvetica, sans-serif;
    list-style-type: none;
    padding: 15px;
  }

.back {
    display: block;
    color: #c00;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 20px;
    font-weight: bold;
    margin-bottom: 20px;
}
.text{
  column-count: 4;
}
@media(max-width: 600px){
  .text{
    column-count: 1;
  }
}

@media(max-width:1000px){
  .text{
    column-count: 2;
  }
}

@media(min-width:1000px){
  .text{
    column-count: 4;
  }
}