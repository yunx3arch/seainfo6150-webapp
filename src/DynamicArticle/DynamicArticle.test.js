import React from "react";
import DynamicArticle from "./DynamicArticle";

describe("DynamicArticle tests", () => {
    it("renders correctly", () => {
        const article = {
                "title": "The Statue of Liberty's torch heads to new museum",
                "authorEmail": "kate.farley@nytimes.com",
                "author": "Kate Farley",
                "timeStamp": "2018-11-22T15:12:24.000Z",
                "displayDate": "November 22nd 2018, 7:12 am",
                "text": "<p>The Statue of Liberty's original torch is getting a new home.</p>",
                "image": {
                    "_url": "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
                }
            };
        const { container } = render(<DynamicArticle article={article}/>);
        expect(container).toMatchSnapshot();
    });
});