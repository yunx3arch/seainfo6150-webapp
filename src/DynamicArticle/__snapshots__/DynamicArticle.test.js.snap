// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicArticle tests renders correctly 1`] = `
<div>
  <div>
    <a
      class="back"
      href="/articlelist"
    >
      Back
    </a>
    <article
      class="article"
    >
      <header>
        <img
          alt="The Statue of Liberty's torch heads to new museum"
          class="image"
          src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
        />
        <h1>
          The Statue of Liberty's torch heads to new museum
        </h1>
        <address>
          by 
          Kate Farley
           (
          <a
            href="mailto:kate.farley@nytimes.com"
          >
            kate.farley@nytimes.com
          </a>
          )
          <br />
        </address>
        <time
          datetime="2018-11-22T15:12:24.000Z"
        >
          November 22nd 2018, 7:12 am
        </time>
      </header>
      <div
        class="text"
      >
        <div>
          <p>
            The Statue of Liberty's original torch is getting a new home.
          </p>
        </div>
      </div>
    </article>
  </div>
</div>
`;
